File: app/src/services/group.service.ts
import { api } from './api';
import type { Character, Conversation, Message } from '@ai-companions/shared';

export const groupService = {
  async createGroup(characterIds: string[]): Promise<Conversation> {
    const response = await api.post('/groups', { character_ids: characterIds });
    return response.data.data;
  },

  async getGroupCharacters(groupId: string): Promise<Character[]> {
    const response = await api.get(`/groups/${groupId}/characters`);
    return response.data.data;
  },

  async addCharacter(groupId: string, characterId: string): Promise<void> {
    await api.post(`/groups/${groupId}/characters`, { character_id: characterId });
  },

  async removeCharacter(groupId: string, characterId: string): Promise<void> {
    await api.delete(`/groups/${groupId}/characters/${characterId}`);
  },

  async sendMessage(groupId: string, content: string): Promise<{ user_message: Message; ai_messages: any[] }> {
    const response = await api.post(`/groups/${groupId}/messages`, { content });
    return response.data.data;
  },
};

File: ai_companions_flutter/lib/core/services/group_service.dart
// Group Service - Group Chat

import 'package:dio/dio.dart';
import '../../core/network/dio_client.dart';
import '../../shared/models/index.dart';

class GroupService {
  GroupService._();

  static final instance = GroupService._();

  Dio get _dio => DioClient.instance.dio;

  Future<Conversation> createGroup(List<String> characterIds) async {
    try {
      final response = await _dio.post(
        '/groups',
        data: {'character_ids': characterIds},
      );
      final data = response.data['data'] as Map<String, dynamic>;
      return Conversation.fromJson(data);
    } catch (e) {
      rethrow;
    }
  }

  Future<List<Character>> getGroupCharacters(String groupId) async {
    try {
      final response = await _dio.get('/groups/$groupId/characters');
      final data = response.data['data'] as List;
      return data.map((e) => Character.fromJson(e as Map<String, dynamic>)).toList();
    } catch (e) {
      rethrow;
    }
  }

  Future<void> addCharacter(String groupId, String characterId) async {
    try {
      await _dio.post(
        '/groups/$groupId/characters',
        data: {'character_id': characterId},
      );
    } catch (e) {
      rethrow;
    }
  }

  Future<void> removeCharacter(String groupId, String characterId) async {
    try {
      await _dio.delete('/groups/$groupId/characters/$characterId');
    } catch (e) {
      rethrow;
    }
  }

  Future<GroupMessageResponse> sendMessage(String groupId, String content) async {
    try {
      final response = await _dio.post(
        '/groups/$groupId/messages',
        data: {'content': content},
      );
      final data = response.data['data'] as Map<String, dynamic>;
      return GroupMessageResponse.fromJson(data);
    } catch (e) {
      rethrow;
    }
  }
}

